### DISCUSSION TITLE
Require advice on setup

### DISCUSSION THREAD

**Discussion by Geekyadz** [opencardev/crankshaft#483] [Q&A]
I'm looking to run Crankshaft on a raspberry Pi 4 8gb version (question: is this overkill?)

I've managed to download Crankshaft OpenCarOS NG Alpha7.2 2021-02-09, however at this stage I am testing using a PC external monitor (not touchscreen) (I'm also toying o the idea of using my TV just for testing as my monitor doesn't have speakers), and finding that it's rebooting every so often.

I've tested AA with my phone (Google Pixel 4a) and it appears to work, but I am unable to do anything with my mouse or keyboard. Has anyone got any advise for this? I feel like I have missed a step here.

I'm looking into put this setup (once it has proven to be stable enough for me) into my Ford Fiesta MK7.

I guess my next question is what screen would you recommend for this? 

Thanks for any help given.

**Comment by exceltek1** [comment]
Hi geekyadz, 
As it turns out I am in the process of building Crankshaft-NG into my VW Amarok and running on a PI4 8GB combined with the official raspberry Pi 7 inch touch screen.  It all seems to be coming together quite nicely. The 7 inch screen is double DIN format so it makes it possible to swapout the old radio / infotainment system as it fits right in place you could go with a larger screen but then this of course requires mounting it all at a convenient position and hiding all the cabling comes with its own challenges.

**Reply by Geekyadz** [reply]
Hi,

Thanks for your reply, personally I'm not looking to replace my stereo; it's got an external screen which also shows messages involving the car, eg when the car is due for a service, parking sensors (if/when it displays. It used to work but since saying that I want the car it appears to have been removed and noone is giving me a straight answer on whether I can get that back or not, though the sensors are still working as I hear the bleeping when I'm near something. There's a whole other things it does as well but I'm not listing absolutely everything). The main part of the stereo is integrated with two vents either side and is slightly curved just to make things even more awkward. So yeah, not really off to a great start on that front. However I'll cross placing it when I know exactly what I have/need. For now I'll happily take a setup that I can remove from the car. Unless of course the software can be made where it can accept messages (which I think are coming from the CANBUS) then I'll think about it.

My main issue is making sure it's stable before looking into getting it into the car. Yesterday's testing didn't seem to go to plan.

It's immediately apparent that I'm gonna need a RTC clock for the main screen. Are any coded into Crankshaft? If so, can you list them please?

I'm also wondering what extra things are on the screen before getting into AA? It didn't seem to load very much, but part of me feels that I'm also needing to get extra kit for that functionally to work.

Yeah, feeling like a complete noob and a little out of my depth.

**Reply by matt2005** [reply]
You should be able to use any rtc as long as its supported by debian buster

**Reply by Geekyadz** [reply]
> You should be able to use any rtc as long as its supported by debian buster

Thanks Matt. Any particular brand you would recommend?

**Comment by Geekyadz** [comment]
> I'm looking to run Crankshaft on a raspberry Pi 4 8gb version (question: is this overkill?)
> 
> I've managed to download Crankshaft OpenCarOS NG Alpha7.2 2021-02-09, however at this stage I am testing using a PC external monitor (not touchscreen) (I'm also toying o the idea of using my TV just for testing as my monitor doesn't have speakers), and finding that it's rebooting every so often.
> 
> I've tested AA with my phone (Google Pixel 4a) and it appears to work, but I am unable to do anything with my mouse or keyboard. Has anyone got any advise for this? I feel like I have missed a step here.
> 
> I'm looking into put this setup (once it has proven to be stable enough for me) into my Ford Fiesta MK7.
> 
> I guess my next question is what screen would you recommend for this?
> 
> Thanks for any help given.

Update:

I've purchased the official 7" pi screen and can report that it seems a lot more stable than hooking it up to my computer monitor, so I guess I never followed the instructions properly. My bad on that one.

I've also tested sounds by plugging a 3.5mm jack into my (household) stereo via phono connections. The volume is quite a lot on the low side. As I'm using the on board jack I'm assuming that it's to do with that and a external DAC will resolve this issue?

It's not the cable as after testing the pi I plugged the jack into my phone and normal volume resumed.

Would be nice to have some suggestions for either products or commands for testing, or an I missing something again as I'm testing when AA it's open? Interestingly, the player before the AA screen is not detecting any music on my phone (which I feel it should be) so I can't even test it to see if it's a AA limitation or if it's the jack being rubbish.

**Comment by Geekyadz** [comment]
![PXL_20210906_190153132](https://user-images.githubusercontent.com/30429165/132254553-a2a1660b-62d3-4e55-8903-151153965761.jpg)

Done a little digging for what hardware I've got.

Found a USB to 3.5mm jack that's used for one of my Corsair headsets (which I don't need and am happy for it to be used on this).

However I'm unable to keep the Corsair output option selected/saved (every time I tap on save it's reverting back to the on board audio apart from the mic which I'm happy to report it just picked it up instantly).

Any suggestions please?

**Comment by Geekyadz** [comment]
Looking for assistance with this file:

![PXL_20210911_162510412 MP](https://user-images.githubusercontent.com/30429165/133128759-5117e28e-38f8-4ab0-8fa4-89502926818c.jpg)

When I ground pin 13 to pin 14 (which is ground) I don't see anything happening with the volume.

Can someone please guide me on how this file works?
